src/main/java/com/epam/ta/reportportal/plugin/Pf4jPluginManager.java

```diff
@@ -137,33 +145,42 @@ public PluginState startUpPlugin(String pluginId) {
 	@Override
 	public boolean loadPlugin(String pluginId, IntegrationTypeDetails integrationTypeDetails) {
 		return ofNullable(integrationTypeDetails.getDetails()).map(details -> {
-			String fileName = IntegrationDetailsProperties.FILE_NAME.getValue(details)
+			String fileName = IntegrationTypeProperties.FILE_NAME.getValue(details)
 					.map(String::valueOf)
 					.orElseThrow(() -> new ReportPortalException(ErrorType.PLUGIN_UPLOAD_ERROR,
 							Suppliers.formattedSupplier("'File name' property of the plugin - '{}' is not specified", pluginId).get()
 					));
 
 			Path pluginPath = Paths.get(pluginsDir, fileName);
 			if (Files.notExists(pluginPath)) {
-				String fileId = IntegrationDetailsProperties.FILE_ID.getValue(details)
+				String fileId = IntegrationTypeProperties.FILE_ID.getValue(details)
 						.map(String::valueOf)
 						.orElseThrow(() -> new ReportPortalException(ErrorType.PLUGIN_UPLOAD_ERROR,
 								Suppliers.formattedSupplier("'File id' property of the plugin - '{}' is not specified", pluginId).get()
 						));
 				try {
-					pluginLoader.copyFromDataStore(fileId, pluginPath);
+					pluginLoader.copyFromDataStore(fileId, pluginPath, Paths.get(resourcesDir, pluginId));
 				} catch (IOException e) {
 					throw new ReportPortalException(ErrorType.PLUGIN_UPLOAD_ERROR,
 							Suppliers.formattedSupplier("Unable to load plugin - '{}' from the data store", pluginId).get()
 					);
 				}
+			} else {
+				try {
+					pluginLoader.copyPluginResource(pluginPath, Paths.get(resourcesDir, pluginId));
+				} catch (IOException e) {
+					throw new ReportPortalException(ErrorType.PLUGIN_UPLOAD_ERROR,
+							Suppliers.formattedSupplier("Unable to load resources of the - '{}' plugin", pluginId).get()
+					);
+				}
 			}
 
 			return ofNullable(pluginManager.loadPlugin(pluginPath)).map(id -> {
 				if (PluginState.STARTED == pluginManager.startPlugin(pluginId)) {
-					Optional<ExtensionPoint> extensionPoint = this.getInstance(pluginId, ExtensionPoint.class);
+					Optional<org.pf4j.ExtensionPoint> extensionPoint = this.getInstance(pluginId, org.pf4j.ExtensionPoint.class);
```

Comment:
it is not pf4j extension point. it's our

Reply:
Old `ExtensionPoint` is a ENUM for storing extension types.
Here was packages misconfiguration - we have to get `pf4j` ExtensionPoint as it's a base interface for our extensions.